tools/rapids-upload-to-s3

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# A utility script that tars up a specified directory and uploads it to S3
+# Positional Arguments:
+#   1) path to tar up
+#   2) s3 path to upload it to
+set -eo pipefail
+
+. rapids-bash-functions
+
+if [ -z "$1" ] || [ -z "$2" ]; then
+  echo_stderr "Must specify both input arguments: PATH_TO_TAR_UP and S3_DEST_PATH"
+  exit 1
+fi
+
+PATH_TO_TAR_UP="$1"
+S3_DEST_PATH="$2"
+
+echo_stderr "Compressing and uploading ${PATH_TO_TAR_UP} to ${S3_DEST_PATH}"
+
+# Tar.gz to stdout and pipe that to the s3 command, no need for a local file
+tar czfv - -C "${PATH_TO_TAR_UP}" . | aws s3 cp --only-show-errors - "${S3_DEST_PATH}"
```

Comment:
If we want to leverage this file for `rapids-upload-misc-to-s3`, we could do something like the following in this script:

- Check `$1` to see if it's a file or a directory
- If it's a file, just copy it as-is to `$2`
- Else, tar it up and then pipe the output to `aws s3 cp ...` like you're already doing.

Up to you. If we adopt this method though, we could consolidate the `Browse parent directory of uploaded file` bits to a single file.